lib/src/generateKernels.cc

```diff
@@ -867,7 +883,11 @@ void genSynapseKernel(const NNmodel &model, //!< Model description
     os << std::endl;
 
     bool firstSynapseGroup = true;
+#ifdef MPI_ENABLE
```

Comment:
The only thing is that: s.second.getPaddedKernelIDRange() is different for localSyapseGroup and Synapsegroup since we should emulate all synapse groups in one node if MPI is not enabled but just emulate part of synapse if MPI is enabled.

Reply:
Well that won't be the case once you get rid of the duplicate groups in the data structures. Perhaps the padded kernel range can be setup correctly so remote groups have no ids assigned to them.